src/decode.rs

```diff
@@ -162,6 +168,62 @@ struct Table {
     depths: Vec<u16>,
 }
 
+/// Describes the static parameters for creating a decoder.
+#[derive(Clone, Debug)]
+pub struct Configuration {
+    order: BitOrder,
+    size: u8,
+    tiff: bool,
+    yield_on_full: bool,
+}
+
+impl Configuration {
+    /// Create a configuration to decode with the specified bit order and symbol size.
+    pub fn new(order: BitOrder, size: u8) -> Self {
+        super::assert_decode_size(size);
+        Configuration {
+            order,
+            size,
+            tiff: false,
+            yield_on_full: false,
+        }
+    }
+
+    /// Create a configuration for a TIFF compatible decoder.
+    pub fn with_tiff_size_switch(order: BitOrder, size: u8) -> Self {
+        super::assert_decode_size(size);
+        Configuration {
+            order,
+            size,
+            tiff: true,
+            yield_on_full: false,
+        }
+    }
+
+    /// Immediately yield to the caller when the decoder buffer is full.
+    ///
+    /// This can be used when dealing with "relaxed" stream interpretation that may not contain an
+    /// explicit EOF but instead expect the decoder to stop fetching symbols when some out-of-band
+    /// specified length of the decoded text has been reached. Symbols afterwards can not be
+    /// expected to be valid. The decoder will yield to the caller instead of returning an error by
+    /// interpreting the following bit-stream.
+    ///
```

Comment:
Might be worth adding a note to the effect of "this matches the behavior of libtiff's LZW decoder", which is the real reason people want it